### PR TITLE
[Notifier] added telegram options

### PR DESCRIPTION
**Pull request by krasilnikovs**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| License       | MIT

I have added `TelegramOptions` model for sending a telegram message with options like the example I have used  `SlackOptions`. 

Testing notes:
1) Create telegram bot https://core.telegram.org/bots#creating-a-new-bot
2) Open telegram and join to the created chat
3) Send a message to telegram
for testing, I used  the scripts 
[scripts.zip](https://github.com/symfony/symfony/files/4499901/scripts.zip)

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ReplyKeyboardMarkup.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @see https://core.telegram.org/bots/api#replykeyboardmarkup
+ */
+class ReplyKeyboardMarkup extends AbstractTelegramReplyMarkup
+{
+    public function __construct()
+    {
+        $this->options['keyboard'] = [];
+    }
+
+    public function keyboard(array $buttons): self
```
Comment (Koc) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ReplyKeyboardMarkup.php:28]: Please provide phpdoc for describe `$buttons` type, like `@param KeyboardButton[] $buttons`

Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ReplyKeyboardMarkup.php:33]: Added

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ */
+abstract class AbstractTelegramReplyMarkup
+{
+    protected $options = [];
+
+    public function toArray(): array
```
Comment (flug) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php:21]: Wouldn't toArray have to be abstract for $options not to conflict with other implementations?

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/InlineKeyboardButton.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup\Button;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ *
+ * @see https://core.telegram.org/bots/api#inlinekeyboardbutton
+ */
+class InlineKeyboardButton extends AbstractKeyboardButton
```
Comment (flug) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/InlineKeyboardButton.php:19]:
```suggestion
final class InlineKeyboardButton extends AbstractKeyboardButton
```

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/KeyboardButton.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup\Button;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ *
+ * @see https://core.telegram.org/bots/api#keyboardbutton
+ */
+class KeyboardButton extends AbstractKeyboardButton
```
Comment (flug) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/KeyboardButton.php:19]:
```suggestion
final class KeyboardButton extends AbstractKeyboardButton
```

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ForceReply.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ *
+ * @see https://core.telegram.org/bots/api#forcereply
+ */
+class ForceReply extends AbstractTelegramReplyMarkup
```
Comment (flug) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ForceReply.php:19]:
```suggestion
final class ForceReply extends AbstractTelegramReplyMarkup
```

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/InlineKeyboardMarkup.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup;
+
+use Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup\Button\InlineKeyboardButton;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ *
+ * @see https://core.telegram.org/bots/api#inlinekeyboardmarkup
+ */
+class InlineKeyboardMarkup extends AbstractTelegramReplyMarkup
```
Comment (flug) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/InlineKeyboardMarkup.php:21]:
```suggestion
final class InlineKeyboardMarkup extends AbstractTelegramReplyMarkup
```

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/AbstractKeyboardButton.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup\Button;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ */
+abstract class AbstractKeyboardButton
+{
+    protected $options = [];
```
Comment (flug) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/AbstractKeyboardButton.php:19]:
```suggestion 
    public abstract function toArray(): array; 
```

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ReplyKeyboardRemove.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ *
+ * @see https://core.telegram.org/bots/api#replykeyboardremove
+ */
+class ReplyKeyboardRemove extends AbstractTelegramReplyMarkup
```
Comment (flug) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ReplyKeyboardRemove.php:19]:
```suggestion
final class ReplyKeyboardRemove extends AbstractTelegramReplyMarkup
```

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Telegram\Reply\Markup;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ */
```
Comment (fabpot) [inline src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php:16]: You should add `@experimental in 5.2` in all new classes.

**Review by fabpot — CHANGES_REQUESTED** [review]
LGTM. Anything else needs to be done before merging?

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ReplyKeyboardRemove.php (hunk 8 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ReplyKeyboardRemove.php]: Added

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/InlineKeyboardMarkup.php (hunk 6 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/InlineKeyboardMarkup.php]: Added

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ForceReply.php (hunk 5 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/ForceReply.php]: Added

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/KeyboardButton.php (hunk 4 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/KeyboardButton.php]: Added

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/InlineKeyboardButton.php (hunk 3 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/InlineKeyboardButton.php]: Added

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php (hunk 2 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php:23]: I don't think that should be abstract

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/AbstractKeyboardButton.php (hunk 7 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/Button/AbstractKeyboardButton.php:21]: I don't think that should be abstract

File: src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php (hunk 9 above)
Comment (krasilnikovm) [reply src/Symfony/Component/Notifier/Bridge/Telegram/Reply/Markup/AbstractTelegramReplyMarkup.php:18]: Added, sorry ma bad.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @krasilnikovm.

**Pull request merged**